engine/src/main/scala/cromwell/engine/workflow/WorkflowStoreActor.scala

```diff
@@ -76,37 +76,35 @@ abstract class WorkflowStore {
   }
 
   def remove(id: WorkflowId): Boolean = {
-    val newWorkflowStore = workflowStore filterNot { _.id == id }
-
-    if (newWorkflowStore == workflowStore) {
-      false
-    } else {
-      workflowStore = newWorkflowStore
+    if (workflowStore.exists(_.id == id)) {
```

Comment:
It just seemed unnecessary to `val newWorkflowStore = workflowStore filterNot { _.id == id }` and compare it to the old one just to check if `id` was indeed inside the workflowStore.